feat(device_info_plus): Add User Device Name in Android

FaizanUllahDev opened the pull request:
## Description

This PR updates the functionality of the `NAME` property to now include and send the customer's device name. This enhancement ensures that the application can capture and utilize the user device name for better identification, personalization, and debugging. Prior to this change, the `NAME` property did not include device-specific details. This update significantly improves logging, user tracking, and system behavior analysis.

## Related Issues

In the current package, the user is unable to obtain the Android user's custom device name

- Fix #3436

## Checklist

- [x] I read the [[Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md)](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the plugin version in `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change
Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a ! in the title as explained in [Conventional Commits] (https://www.conventionalcommits.org/en/v1.0.0)).
- [x]  this is not a breaking change.

miquelbeltran commented:
Hello, can you fix the markdown of the PR description, as well as adding the plugin name between `( )` on the PR title?

The feature seems good to me, I'll take a deeper look when I got the time.

miquelbeltran commented:
The title format is still not right, please read the contributor guidelines and look at other PRs on how it is done

miquelbeltran commented:
thanks, although I don't think this is a breaking change

FaizanUllahDev commented:
Yes! but this change will save the time and give possibles user name of the current user device.

miquelbeltran commented:
I've been working today on this PR and found several things that need changes, however when I tried to push my changes I've got remote rejected. Can you enable that we can push to your branch?

miquelbeltran commented on packages/device_info_plus/device_info_plus/android/src/main/kotlin/dev/fluttercommunity/plus/device_info/DeviceInfoPlusPlugin.kt:
```diff
@@ -24,8 +24,9 @@ class DeviceInfoPlusPlugin : FlutterPlugin {
-        val handler = MethodCallHandlerImpl(packageManager, activityManager)
+        val activityManager: ActivityManager =
+            context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+        val handler = MethodCallHandlerImpl(packageManager, activityManager, context)
```
Don't pass `context` here, instead pass the `ContentResolver`.

miquelbeltran commented on packages/device_info_plus/device_info_plus/android/src/main/kotlin/dev/fluttercommunity/plus/device_info/MethodCallHandlerImpl.kt:
```diff
@@ -37,6 +40,7 @@ internal class MethodCallHandlerImpl(
             build["manufacturer"] = Build.MANUFACTURER
             build["model"] = Build.MODEL
             build["product"] = Build.PRODUCT
+            build["name"] = Settings.Global.getString(context.contentResolver, Settings.Global.DEVICE_NAME)
```
This API is only available from Android N (API 25), this code needs to check for the version. Android Studio should show a warning.

miquelbeltran commented on packages/device_info_plus/device_info_plus/lib/src/model/android_device_info.dart:
```diff
@@ -89,6 +90,10 @@ class AndroidDeviceInfo extends BaseDeviceInfo {
   final String product;
 
+  /// The name of the device.
+  /// https://developer.android.com/reference/android/os/Build#NAME
```
this documentation is incorrect, it should link to `https://developer.android.com/reference/android/provider/Settings.Global#DEVICE_NAME`

FaizanUllahDev commented:
> I've been working today on this PR and found several things that need changes, however when I tried to push my changes I've got remote rejected. Can you enable that we can push to your branch?

Please check now I set the bypass rule for the update

miquelbeltran commented:
It won't let me push, probably because the PR was created from `main` instead of a branch? anyway I added comments on what needs to be changed

FaizanUllahDev commented:
> It won't let me push, probably because the PR was created from `main` instead of a branch? anyway I added comments on what needs to be changed

I added you as contributor so, I think now you can do changes

miquelbeltran commented:
ugh what just happened, I pushed and it closed the PR?

miquelbeltran commented:
Reopening

miquelbeltran commented:
Well, I don't know. Git is giving me lots of problems, opening a pr from `main` directly can cause issues. I pushed the code on a new branch in your repo. Create a new PR following this link: https://github.com/fluttercommunity/plus_plugins/compare/main...FaizanUllahDev:plus_plugins:3437?expand=1 and copy the description from this PR.
